optimum/intel/neural_compressor/quantization.py

```diff
@@ -35,101 +40,292 @@
+            task (`str`, defaults to None):
+                The task defining the model topology used for the ONNX export.
+            seed (`int`, defaults to 42):
+                The random seed to use when shuffling the calibration dataset.
         """
+        super().__init__()
+        self.model = model
+        self.eval_fn = eval_fn
+        self.task = task
+        self.seed = seed
+        signature = inspect.signature(self.model.forward)
+        self._signature_columns = list(signature.parameters.keys())
+        self.input_names = None
 
-        self.config = config.config if isinstance(config, IncQuantizationConfig) else Quantization_Conf(config)
-        self.approach = IncQuantizationMode(self.config.usr_cfg.quantization.approach)
-        self.eval_func = eval_func
-        self.train_func = train_func
-        self.calib_func = calib_func
-        if calib_dataloader is not None:
-            calib_dataloader = IncDataLoader.from_pytorch_dataloader(calib_dataloader)
-        self.calib_dataloader = calib_dataloader
-
-        if self.config.usr_cfg.model.framework == "pytorch_fx":
-            neural_compressor.adaptor.pytorch._cfgs_to_fx_cfgs = _cfgs_to_fx_cfgs
-
-        self.quantization = Quantization(self.config)
-
-        self.quantization.eval_func = self.eval_func
-
-        if self.approach == IncQuantizationMode.STATIC:
-            if self.calib_func is not None:
-                self.quantization.calib_func = self.calib_func
-            if self.calib_dataloader is not None:
-                self.quantization._calib_dataloader = self.calib_dataloader
-
-        if self.config.usr_cfg.model.framework == "pytorch_ipex":
-            raise ValueError("INC IPEX only is not currently supported.")
-
-        if self.approach == IncQuantizationMode.AWARE_TRAINING:
-            if self.train_func is None:
-                raise ValueError("train_func must be provided for quantization aware training.")
-            self.quantization.q_func = self.train_func
-            if not is_torch_less_than_1_13:
-                if self.calib_dataloader is None:
-                    raise ValueError(
-                        "For quantization aware training, a calibration dataloader `calib_dataloader` must be provided for PyTorch 1.13 or above."
-                    )
-                self.quantization._calib_dataloader = self.calib_dataloader
+    @classmethod
+    def from_pretrained(cls, model: PreTrainedModel, **kwargs):
+        # TODO : Create model
+        return cls(model, **kwargs)
+
+    def quantize(
+        self,
+        save_directory: Union[str, Path],
+        quantization_config: "PostTrainingQuantConfig",
+        calibration_dataset: Dataset = None,
+        file_name: Optional[str] = None,
+        batch_size: int = 8,
+        data_collator: Optional[DataCollator] = None,
+        remove_unused_columns: bool = True,
```

Comment:
sorry, I didn't get the point of this parameter. could you elaborate more?

Reply:
It gives the possibility to remove or not the columns of the dataset unused by the model forward method